google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -46,59 +50,82 @@ def format_stackdriver_json(record, message):
     return json.dumps(payload)
 
 
-def get_trace_id_from_flask():
-    """Get trace_id from flask request headers.
+def get_request_data_from_flask():
+    """Get trace_id and http_request data from flask request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        dict: data about the associated http request.
     """
     if flask is None or not flask.request:
-        return None
+        return None, None
+
+    # build http_request
+    http_request = {
+        "request_method": flask.request.method,
+        "request_url": flask.request.url,
+        "request_size": str(flask.request.content_length),
+        "user_agent": flask.request.user_agent.string,
+        "remote_ip": flask.request.remote_addr,
+        "referer": flask.request.referrer,
+    }
 
+    # find trace id
+    trace_id = None
     header = flask.request.headers.get(_FLASK_TRACE_HEADER)
+    if header:
+        trace_id = header.split("/", 1)[0]
```

Comment:
Future edge case to take care of: 

Given user input is `X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE` where any of the values can be nil or 0. This logic to split by / and grab the first value might give you `span id` instead. 

Seems super niche edge case, but I actually got a similar ticket for this in Go.

Reply:
Hmm I'm not sure I understand. Can you have a span without a trace? Do you have a ticket or doc I can look over?

Either way, I'm not a huge fan of this parsing logic. I'm hoping to fix it up when I add support for spans soon

Reply:
It shouldn't be possible to have a trace as null but span as non-null. Curious how this was a case in Go?

Reply:
The edge cases I'm referring to here is just that the `X-Cloud-Trace-Context` field is provided by the user. 

So the user could manually write anything here. e.g. `'/hello/world'` and we'd grep the first token as trace_id. Adding better regex later on could help.

[This](https://github.com/googleapis/google-cloud-go/issues/1590) was the original Go issue.

Reply:
Where we tell users they can manually write this header: https://cloud.google.com/trace/docs/setup#force-trace

Reply:
Hmm good to keep in mind, yeah it might be helpful to throw an exception or something if the header doesn't match the expected format

I'll probably fix that in a later PR when I add support for spans though. That's when proper parsing will be more important. For now, this is the logic that was already in place